x-pack/plugins/observability_solution/inventory/public/pages/inventory_page/index.tsx

```diff
@@ -62,47 +34,10 @@ export function InventoryPage() {
     return () => {
       searchBarContentSubscription.unsubscribe();
     };
-  });
-
-  function handlePageChange(nextPage: number) {
-    inventoryRoute.push('/', {
-      path: {},
-      query: { ...query, pageIndex: nextPage },
-    });
-  }
-
-  function handleSortChange(sorting: EuiDataGridSorting['columns'][0]) {
-    inventoryRoute.push('/', {
-      path: {},
-      query: {
-        ...query,
-        sortField: sorting.id as EntityColumnIds,
-        sortDirection: sorting.direction,
-      },
-    });
-  }
-
-  function handleTypeFilter(entityType: EntityType) {
-    inventoryRoute.push('/', {
-      path: {},
-      query: {
-        ...query,
-        // Override the current entity types
-        entityTypes: [entityType],
-      },
-    });
-  }
+    // If query has updated, the inventoryRoute state is also updated
+    // as well, so we only need to track changes on query.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [query, refreshSubject$, searchBarContentSubject$]);
 
-  return (
-    <EntitiesGrid
-      entities={value.entities}
-      loading={loading}
-      sortDirection={sortDirection}
-      sortField={sortField}
-      onChangePage={handlePageChange}
-      onChangeSort={handleSortChange}
-      pageIndex={pageIndex}
-      onFilterByType={handleTypeFilter}
-    />
-  );
+  return groupBy === 'unified' ? <UnifiedInventory /> : <GroupedInventory />;
```

Comment:
Based on the comment above:

1. 
```suggestion
  return groupBy === 'none' ? <UnifiedInventory /> : <GroupedInventory />;
```

2. 
```suggestion
  return entityView === 'unified' ? <UnifiedInventory /> : <GroupedInventory />;
```